cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -53,7 +53,12 @@ function main {
     echo "Adding entry to fstab: $fstab"
     echo "$fstab" >> /etc/fstab
     echo -n "Mounting: $device to $mount_location"
-    mount "$mount_location"
+    until grep ^$device /etc/mtab > /dev/null; do sleep 1; echo -n .; mount "$mount_location" || :; done
+    if [ "$mount_location" = "/var/log" ]; then
+      echo "Recreating /var/log/journal"
+      mkdir /var/log/journal
+      systemd-tmpfiles --create --prefix /var/log/journal
+    fi
```

Comment:
Good catch! I'm a bit hesitant to just remove `-e` though.
Maybe we could wrap the echo in a function so that errexit is turned off while printing to stdout. But then what if it happens during execution of one of the other commands that output something.

I'll have to think a bit more about it with a refreshed mind tomorrow (almost midnight here).